### PR TITLE
OV-204: Fetch voices and save them to store

### LINKED ISSUE #204: FEAT: Fetch voices and save them to store
### What feature?

When user opens `scripts` tab on video-menu, a list of voices must be fetched and saved to store. When request is in progress show loader.

Use branch from [azure text to speech pr](https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/185) as your base branch.

### Add screenshots

_No response_

### Acceptance criteria

_No response_

### PR DESCRIPTION
**Pull request by JKaypa**
Here we fetch voices and show the loader only the first time we open the script tab.

### CODE REVIEW

**Review by Alblupynos — DISMISSED** [review]

File: frontend/src/bundles/studio/store/slice.ts [hunk 1]
```diff
@@ -61,6 +67,10 @@ type State = {
     scenes: Array<Scene>;
     scripts: Array<Script>;
     videoSize: VideoPreviewT;
+    voices: {
+        dataStatus: ValueOf<typeof DataStatus>;
+        items: Voice[] | [];
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/store/slice.ts:72]: You don't need this union. Empty array is also acceptable for Voice typed array

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/studio/store/slice.ts [hunk 2]
```diff
@@ -79,6 +89,10 @@ const initialState: State = {
     scenes: [{ id: uuidv4(), duration: MIN_SCENE_DURATION }],
     scripts: [],
     videoSize: VideoPreview.LANDSCAPE,
+    voices: {
+        dataStatus: DataStatus.IDLE,
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/store/slice.ts:93]: We should have only one dataStatus for slice like it was mentioned in discussion https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/236#discussion_r1751960832

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx [hunk 3]
```diff
@@ -18,7 +21,14 @@ import { Script, VoicesModal } from './components/components.js';
 
 const ScriptContent: React.FC = () => {
     const dispatch = useAppDispatch();
-    const scripts = useAppSelector(({ studio }) => studio.scripts);
+    const studio = useAppSelector(({ studio }) => studio);
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx:24]:
```suggestion
    const { voices, scripts, dataStatus } = useAppSelector(({ studio }) => studio);
```

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx [hunk 4]
```diff
@@ -18,7 +21,14 @@ import { Script, VoicesModal } from './components/components.js';
...
+
+    useEffect(() => {
+        if (studio.voices.length === 0) {
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx:27]:
```suggestion
        if (!voices.length) {
```

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx [hunk 5]
```diff
@@ -38,26 +48,36 @@ const ScriptContent: React.FC = () => {
     return (
         <>
             <VStack w="full" spacing="20px" p="20px 0">
-                {scripts.length === 0 ? (
-                    <Text variant="body1" width="60%" textAlign="center">
-                        To add a script press a button below.
-                    </Text>
+                {studio.dataStatus === DataStatus.PENDING ? (
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx:51]:
```suggestion
                {dataStatus === DataStatus.PENDING ? (
```

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx [hunk 6]
```diff
@@ -38,26 +48,36 @@ const ScriptContent: React.FC = () => {
...
+                        {studio.scripts.length === 0 ? (
+                            <Text
+                                variant="body1"
+                                width="60%"
+                                textAlign="center"
+                            >
+                                To add a script press a button below.
+                            </Text>
+                        ) : (
+                            studio.scripts.map(({ id, ...script }) => (
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx:64]:
```suggestion
                            scripts.map(({ id, ...script }) => (
```

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx (hunk 4 above)
Comment (JKaypa) [reply frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx:27]: Yes, I also tried to do that, but ESLint doesn't like that way

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx [hunk 7]
```diff
@@ -38,26 +48,36 @@ const ScriptContent: React.FC = () => {
...
+                    <Loader />
                 ) : (
-                    scripts.map(({ id, ...script }) => (
-                        <Script
-                            key={id}
-                            id={id}
-                            {...script}
-                            handleChangeVoice={handleChangeVoiceClick}
+                    <>
+                        {studio.scripts.length === 0 ? (
```
[the remark this answers is not in the document]
Comment (JKaypa) [reply frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/script-content.tsx:55]: Yes, I also tried to do that, but ESLint doesn't like that way

**Review by anton-otroshchenko — DISMISSED** [review]

**Pull request merged**